src/Docfx.Common/IItemWithMetadata.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.\r
+// The .NET Foundation licenses this file to you under the MIT license.
+
+
+namespace Docfx.Common;
+
+
+/// <summary>
+///   An item that contains metadate
+/// </summary>
+public interface IItemWithMetadata
```

Comment:
Consider changing this to an internal interface. We could use `InternalsVisibleTo` to make `Docfx.Common` visible to `Docfx.Dotnet`.

Reply:
If I do this, I have plenty of compiler errors, like:
```text
12>ApiPage.cs(12,8): Error CS0656 : Missing compiler required member 'System.Runtime.CompilerServices.RequiredMemberAttribute..ctor'
12>ApiPage.cs(12,8): Error CS0656 : Missing compiler required member 'System.Runtime.CompilerServices.CompilerFeatureRequiredAttribute..ctor'
```

Reply:
On my environment.
It can successfully build with following steps.

1. Checkout PR branch  with following command.
> gh pr checkout 9937
2. Add following settings at `Docfx.Common.csproj`
```
  <ItemGroup>
    <InternalsVisibleTo Include="Docfx.Build.ManagedReference" />
    <InternalsVisibleTo Include="Docfx.Dotnet" />
  </ItemGroup>
```
3. Run `dotnet build`

---
`ApiPage.cs(12,8):` errors seems to be occurred by PR #9960. (It's not merged to this branch yet thought)

If it's occurred when merging latest `main` branch code.
Try clean rebuild.

And if error still continues.
Please attach build log file (`msbuild.binlog`)  that is generated by following command.

> dotnet build --binaryLogger /p:ReportAnalyzer=true

Reply:
To resolve CI errors.
Try to merge latest `main`. and resolve `Docfx.Common.csproj` conflicts with following modifications.

```
  <!-- TODO: Following settings will be removed after NewtonsoftJson dependencies are removed. -->
  <ItemGroup>
    <InternalsVisibleTo Include="docfx.Build" />
    <InternalsVisibleTo Include="Docfx.Build.Tests" />
  </ItemGroup>
  
  <ItemGroup>
    <InternalsVisibleTo Include="Docfx.Build.ManagedReference"/>
    <InternalsVisibleTo Include="Docfx.Dotnet"/>
  </ItemGroup>
```